### PR TITLE
Serialize

### PR DESCRIPTION
**Pull request by agoulti**
* Adding simple wildcard support. So far only a single single asterisk on the lowest level of the path is supported.

* StorageObject is not Serializable, which caused problems when attempting to execute on a slave.

* Added logging to the listener rather than the info log. This way these messages show up in the console.

### CODE REVIEW

File: src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java [hunk 1]
```diff
@@ -189,18 +189,20 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace,
...
 
-      DownloadSpec spec = new DownloadSpec(dirPath, objects);
+      DownloadSpec spec = new DownloadSpec(objects);
       // TODO(agoulti): add a download report.
 
       String resolvedPrefix = StorageUtil
           .replaceMacro(pathPrefix, run, listener);
 
-      initiateDownloadsAtWorkspace(getCredentials(), run, spec, dirPath,
+      initiateDownloadsAtWorkspace(getCredentials(), run, spec, dirPath, listener,
```
Comment (bwkimmel) [inline src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java]: This line (plus a few others in this file) generates a checkstyle error (longer than 80 chars).

File: src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java [hunk 2]
```diff
@@ -306,19 +309,38 @@ public void checkRoles(RoleChecker checker)
...
+    }
+
+    private final String bucket;
+    private final String name;
+  }
+
+  /**
    * DownloadSpec is a way to encode what needs to be downloaded.
    */
   protected static class DownloadSpec implements Serializable {
```
Comment (bwkimmel) [inline src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java]: Is this class needed anymore? It looks like you could just pass around the list of StorageObjectIds.

File: src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java (hunk 1 above)
Comment (agoulti) [reply src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java]: Yes, thanks. It seems Intellij has lost my settings over a restart :-(

File: src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java (hunk 2 above)
Comment (agoulti) [reply src/main/java/com/google/jenkins/plugins/storage/DownloadStep.java]: Good point

**Pull request merged**